src/Raven.Server/RavenServerStartup.cs

```diff
@@ -157,23 +159,90 @@ private async Task RequestHandler(HttpContext context)
                 context.Response.StatusCode = (int)HttpStatusCode.OK;
                 context.Response.Headers["Content-Type"] = "application/json; charset=utf-8";
                 context.Response.Headers[Constants.Headers.ServerVersion] = RavenVersionAttribute.Instance.AssemblyVersion;
+                // Check context url AND TW clients
+                bool isQueryBool = (TrafficWatchManager.HasRegisteredClients && IsQuery(context));
+
+                if (isQueryBool)
+                {
```

Comment:
try to wrap this inside dedicated httpcontext that will handle all of this additional stuff